pegen/pegen.c

```diff
@@ -2,6 +2,62 @@
 #include "pegen.h"
 #include "v38tokenizer.h"
 
+int
+raise_syntax_error(Parser *p, const char *errmsg, ...)
+{
+    // TODO: comvert from bytes offset to character offset
+    PyObject *value = NULL;
+    PyObject *errstr = NULL;
+    PyObject *loc = NULL;
+    PyObject *tmp = NULL;
+    PyObject* filename = NULL;
+    Token *t = p->tokens[p->fill - 1];
+    va_list va;
+
+    va_start(va, errmsg);
+    errstr = PyUnicode_FromFormatV(errmsg, va);
+    va_end(va);
+    if (!errstr) {
+        goto error;
+    }
+    if (filename) {
```

Comment:
How can this condition be true? It's initialized to NULL and not set between there and here. I'm guessing you meant `if (p->tok->filename)` ? (And could that be `Py_None`? Could `p->tok` be NULL?)

Reply:
Yeah, that was meant to be `(p->tok->filename)`.

> Could p->tok be NULL?

The previous code was using `p->tok` directly is a similar fashion so I will assume at this point it won't be true, but we can add some early exit or asserts. What do you prefer?

Reply:
No need to add checks in that case.